OrcScala/src/orc/run/core/Token.scala

```diff
@@ -849,11 +922,18 @@ private class LongCounter(private var value: Long) {
 }
 
 object Token {
+  // MULTI_SCHED_DEBUG
+  /*
+  private val isRunningAlreadyCleared = new ThreadLocal[Boolean]() {
+    override def initialValue() = false
+  }
+  */
+
   private val currentTokenDebugId = new ThreadLocal[LongCounter]() {
     override def initialValue() = new LongCounter(0L)
   }
   def getNextTokenDebugId(runtime: OrcRuntime): Long =
-    /* FIXME:This adverse coupling to runtime should be removed */
+    /* FIXME:This adverse coupling to runtime should be removed. Why not add a runtimeThreadId to the Orc trait? */
```

Comment:
I don't agree. I think logging utilities are very reasonable to be there. Orc represents access to an Orc implementation not some abstract idea of an Orc runtime. If you wanted we could have an abstract interface that is outward facing and one with details like this that is used throughout the implementation tree.